vllm/worker/model_runner.py

```diff
@@ -244,19 +250,27 @@ def _prepare_prompt(
             dtype=torch.int,
             device=self.device,
         )
-        start_loc_tensor = torch.arange(0,
-                                        len(prompt_lens) * max_prompt_len,
-                                        max_prompt_len,
-                                        dtype=torch.long,
-                                        device=self.device)
         prompt_lens_tensor = torch.tensor(prompt_lens,
                                           dtype=torch.long,
                                           device=self.device)
 
+        # Cumulative index of each prompt. [prompt_lens + 1]
+        # [0, 0+1th, 0+1th+2nd, ...]
+        start_loc_tensor = torch.zeros(prompt_lens_tensor.shape[0] + 1,
```

Comment:
@WoosukKwon I think this is wrong after talking with @LiuXiaoxuanPKU , but somehow context attention still works with this. (it should be subquery lens)

Do we have an e2e test for this?